tpp-app/src/settings/BackUpAccount.js

```diff
@@ -0,0 +1,154 @@
+import React, {useEffect, useState} from 'react';
+import {View, Text, StyleSheet, ImageBackground, TouchableOpacity, Alert} from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import OnboardingBackground from '../../ios/tppapp/Images.xcassets/SplashScreenBackground.imageset/colourwatercolour.png'
+import {appleAuth, AppleButton} from '@invertase/react-native-apple-authentication';
+import {
+    GETAppleUser,
+    GETLastSavedBackupTime,
+    POSTBackupToiCloud
+} from "../services/AppleService";
+import {onAppleButtonPress} from "../onboarding/AppleSignin";
+
+export default function BackUpAccount ({ navigation }) {
+    const [isSignedIn, setIsSignedIn] = useState(false);
+    const [lastBackupDateTime, setLastBackupDateTime] = useState("");
+
+    useEffect(() => {
+        // Check if user is currently logged into an Apple account
+        async function checkUserCredentialState() {
+            // get current authentication state for user
+            const user = await GETAppleUser();
+            console.log(user);
+            // /!\ This method must be tested on a real device. On the iOS simulator it always throws an error.
+            const credentialState = await appleAuth.getCredentialStateForUser(user);
```

Comment:
do u know why?

Reply:
hmm not sure Alex actually wrote the sign in part but it's copied from the (docs)[https://github.com/invertase/react-native-apple-authentication]

i think he did some digging and the appleAuth is not super compatible with react-native on simulators (Apple or Facebook, this is on one of u 😤 ) so it only works when its running on a real device

Reply:
I tried for a ridiculously long time before i realized it doesn't work on simulators but it does work on my actual physical phone.

Reply:
![image](https://user-images.githubusercontent.com/35851484/166011618-fccfb9c9-9870-46a9-9751-fae1b63a9fde.png)

Reply:
https://developer.apple.com/forums/thread/651533

Reply:
https://developer.apple.com/forums/thread/651533